File: n349-intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    var ret = []
    
    for (let n1 of nums1) {
        if (nums2.includes(n1) && ! ret.includes(n1)) {
            ret.push(n1)
        }
    }
    
    return ret
};

File: n349-intersection-of-two-arrays.go
package main

func intersection(nums1 []int, nums2 []int) []int {
	data := map[int]int{}
	ret := []int{}

	for _, num1 := range nums1 {
		for _, num2 := range nums2 {
			if num2 == num1 {
				if _, exists := data[num1]; !exists {
					ret = append(ret, num1)
					data[num1] = 1
				}
			}
		}
	}

	return ret
}
